src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -243,17 +245,26 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
 
                         var location = token.GetLocation();
                         var symbolUsageInfo = GetSymbolUsageInfo(token.Parent, semanticModel, syntaxFacts, semanticFacts, cancellationToken);
+
                         var isWrittenTo = symbolUsageInfo.IsWrittenTo();
                         locations.Add(new FinderLocation(token.Parent, new ReferenceLocation(
                             document, alias, location, isImplicit: false,
-                            symbolUsageInfo, candidateReason: match.reason)));
+                            symbolUsageInfo, GetAdditionalProperties(token.Parent, semanticModel, syntaxFacts), candidateReason: match.reason)));
                     }
                 }
             }
 
             return locations.ToImmutableAndFree();
         }
 
+        private static ImmutableArray<AdditionalProperty> GetAdditionalProperties(SyntaxNode node, SemanticModel semanticModel, ISyntaxFactsService syntaxFacts)
+        {
+            var additionalProperties = new ArrayBuilder<AdditionalProperty>();
+            additionalProperties.Add(GetContainingTypeInfo(node, semanticModel, syntaxFacts));
+            additionalProperties.Add(GetContainingMemberInfo(node, semanticModel, syntaxFacts));
```

Comment:
these functions can return null.  So AdditionalProperties will contain null values.  We should avoid that.